sdk/src/signature.rs

```diff
@@ -25,6 +28,49 @@ const MAX_BASE58_SIGNATURE_LEN: usize = 88;
 #[derive(Serialize, Deserialize, Clone, Copy, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub struct Signature(GenericArray<u8, U64>);
 
+impl proptest::prelude::Arbitrary for Signature {
+    type Parameters = ();
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        // Create a strategy that generates 64 random bytes and turns them into a Signature
+        any::<[u8; 64]>()
+            .prop_map(|bytes| Signature(GenericArray::clone_from_slice(&bytes)))
+            .boxed()
+    }
+
+    type Strategy = proptest::prelude::BoxedStrategy<Signature>;
+}
+
+// Implement Arbitrary for Signature
+impl<'a> arbitrary::Arbitrary<'a> for Signature {
```

Comment:
Just a Rust language question.. was the lifetime annotation required here? I would've thought this was a case where it was just implicit.

Reply:
It was required by trait definition of arbitrary::Arbitrary